test/e2e/specs/editor/various/embedding.spec.js

```diff
@@ -0,0 +1,310 @@
+		).toHaveAttribute( 'title', 'Embedded content from twitter' );
+
+		await embedUtils.insertEmbed(
+			'https://twitter.com/wooyaygutenberg123454312'
+		);
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'Valid provider; invalid content. Should render failed, edit state.'
+		).toHaveValue( 'https://twitter.com/wooyaygutenberg123454312' );
+
+		await embedUtils.insertEmbed(
+			'https://wordpress.org/gutenberg/handbook/'
+		);
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'WordPress invalid content. Should render failed, edit state.'
+		).toHaveValue( 'https://wordpress.org/gutenberg/handbook/' );
+
+		await embedUtils.insertEmbed( 'https://twitter.com/thatbunty' );
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'Provider whose oembed API has gone wrong. Should render failed, edit state.'
+		).toHaveValue( 'https://twitter.com/thatbunty' );
+
+		await embedUtils.insertEmbed(
+			'https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/'
+		);
+		await expect(
+			currenEmbedBlock.locator( 'figure' ),
+			'WordPress valid content. Should render valid figure element.'
+		).toHaveClass( 'wp-block-embed' );
+
+		await embedUtils.insertEmbed(
+			'https://www.youtube.com/watch?v=lXMskKTw3Bc'
+		);
+		await expect(
+			currenEmbedBlock.locator( 'figure' ),
+			'Video content. Should render valid figure element, and include the aspect ratio class.'
+		).toHaveClass( /wp-embed-aspect-16-9/ );
+
+		await embedUtils.insertEmbed( 'https://cloudup.com/cQFlxqtY4ob' );
+		await expect(
+			currenEmbedBlock.locator( 'iframe' ),
+			'Photo content. Should render valid iframe element.'
+		).toHaveAttribute( 'title', 'Embedded content from cloudup' );
+	} );
+
+	test( 'should allow the user to convert unembeddable URLs to a paragraph with a link in it', async ( {
+		editor,
+		embedUtils,
+	} ) => {
+		await embedUtils.interceptRequests( {
+			'https://twitter.com/wooyaygutenberg123454312':
+				MOCK_CANT_EMBED_RESPONSE,
+		} );
+		await embedUtils.insertEmbed(
+			'https://twitter.com/wooyaygutenberg123454312'
+		);
+
+		const embedBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Embed',
+		} );
+
+		await expect( embedBlock ).toContainText(
+			'Sorry, this content could not be embedded.'
+		);
+
+		await embedBlock
+			.getByRole( 'button', { name: 'Convert to link' } )
+			.click();
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content:
+						'<a href="https://twitter.com/wooyaygutenberg123454312">https://twitter.com/wooyaygutenberg123454312</a>',
+				},
+			},
+		] );
+	} );
+
+	// Reason: A possible regression of https://github.com/WordPress/gutenberg/pull/14705.
+	test.skip( 'should retry embeds that could not be embedded with trailing slashes, without the trailing slashes', async ( {
```

Comment:
This test is currently failing. When I try to test this scenario manually, it doesn't work at all. I'm skipping the test for now, and we can enable it when regression is fixed.

Reply:
I also confirmed that retry doesn't work. Should we open a new issue as a regression?

Reply:
Yes, let's open the new issue and reference this PR.

Reply:
Thank you, @t-hamano 🙏